apps/common/behaviours/resume-form-session.js

```
@@ -0,0 +1,85 @@
+      } else {
+        // If there are no saved applications for user remove radio option for continue an application
+        const applicationTypeOptions = req.form.options.fields['application-form-type'].options;
+        req.form.options.fields['application-form-type'].options = applicationTypeOptions.filter(
+          radio => radio.value !== 'continue-an-application'
+        );
+      }
+      req.sessionModel.set('applicant-id', applicantId);
+      req.sessionModel.set('licence-type', licenceType);
+    } catch (error) {
+      req.log('error', `Failed to get saved application: ${genAxiosErrorMsg(error)}`);
+      return next(error);
+    }
+    return super.getValues(req, res, next);
+  }
+
+  cleanSession(req) {
+    const { sessionDefaults } = config;
+    const sessionAttributes = Object.keys(req.sessionModel.attributes);
+    const cleanList = sessionAttributes.filter(item => !sessionDefaults.fields.includes(item));
+    req.sessionModel.unset(cleanList);
+  }
+
+  saveValues(req, res, next) {
+    const resumeApplication = req.form.values['application-form-type'] === 'continue-an-application';
+    const applicationToResume = req.sessionModel.get('application-to-resume');
+    if (resumeApplication && applicationToResume) {
+      this.resumeSession(req, applicationToResume);
+    }
+    req.sessionModel.unset('application-to-resume');
+    return super.saveValues(req, res, next);
+  }
+
+  resumeSession(req, application) {
+    req.log('info', `Resuming Form Session: ${application.id}`);
+
+    let session;
+
+    const savedApplicationProps = {
+      'application-id': application.id,
+      'application-created-at': application.created_at,
+      'application-expires-at': application.expires_at
+    };
+
+    try {
+      session = JSON.parse(application.session);
+    } catch (error) {
+      // Log the error
+      req.log('warn', `Error parsing session: ${error}`);
+      session = application.session;
```

maybe throw an error with something like "couldn't restore saved application" instead of restoring potentially corrupted data